File: solutions/free-pascal/armstrong-numbers/1/ArmstrongNumbers.pas
unit ArmstrongNumbers;

{$mode ObjFPC}{$H+}

interface

function IsArmstrongNumber(const number: qword) : boolean;

implementation

uses SysUtils, Math;

function IsArmstrongNumber(const number: qword) : boolean;
var 
  numstr : string;
  c : char;
  len, sum : integer;
begin
  numstr := IntToStr(number);
  len := length(numstr);
  sum := 0;
  for c in numstr do
    sum := sum + round(intpower(StrToInt(c), len));
  result:= (sum = number);
end;

end.

File: solutions/dart/armstrong-numbers/2/lib/armstrong_numbers.dart
import 'dart:math';

class ArmstrongNumbers {
  bool isArmstrongNumber(String number){
    final length = number.length;
    var sum = number.codeUnits.fold(BigInt.zero, (s,d) =>
                         s + BigInt.from(d-48).pow(length));
    return BigInt.parse(number) == sum;
  }
}
